selfdrive/sensord/sensors_qcom2.cc

```diff
@@ -148,33 +124,31 @@ int sensor_loop() {
 
   // Initialize sensors
   std::vector<Sensor *> sensors;
-  for (auto &sensor : sensors_init) {
-    int err = sensor.first->init();
+  for (auto &[sensor, required] : sensors_init) {
+    int err = sensor->init();
     if (err < 0) {
-      // Fail on required sensors
-      if (sensor.second) {
+      if (required) {
         LOGE("Error initializing sensors");
-        delete i2c_bus_imu;
         return -1;
       }
     } else {
-      if (sensor.first == &bmx055_magn || sensor.first == &mmc5603nj_magn) {
+
+      if (sensor == &bmx055_magn || sensor == &mmc5603nj_magn) {
         has_magnetometer = true;
       }
 
-      if (!sensor.first->has_interrupt_enabled()) {
-        sensors.push_back(sensor.first);
+      if (!sensor->has_interrupt_enabled()) {
+        sensors.push_back(sensor);
       }
     }
   }
 
   if (!has_magnetometer) {
     LOGE("No magnetometer present");
-    delete i2c_bus_imu;
     return -1;
   }
 
-  PubMaster pm({"sensorEvents"});
+  PubMaster pm({PM_GYRO, PM_GYRO_2, PM_ACCEL, PM_ACCEL_2, PM_TEMP, PM_LIGHT, PM_MAGN});
```

Comment:
no need to define the socket names as constants

Reply:
thought its more nice this way, cause we use this constants in the sensors too^^

Reply:
done